doc/src/build/programming-with-objects/ch3-immutable-objects.md

```diff
@@ -0,0 +1,152 @@
+## Chapter 3: Immutable Objects
+In chapters 1 and 2, we learned how to create and use objects owned by an account address. In this chapter, we will demonstrate how to create and use immutable objects.
+
+Objects in Sui can have different types of [ownership](../objects.md#object-ownership), with two broad categories: owned objects and shared objects. And there are two types of shared objects: shared immutable and shared mutable. A shared immutable object is an object that can **never** be mutated, transferred or deleted. Because of this immutability, the object is not owned by anyone, and hence it can be used by anyone (thus shared). We will call this type of object an "*immutable object*" for short. In a latter chapter, we discuss shared mutable objects, which is what we call *shared objects* for short.
```

Comment:
I find all of this phrasing super confusing. It is really hard for me to follow if you say, We have "shared immutable" and "shared mutable", but we will just call them "immutable" and "shared" for short. 

I found this doubly confusing while reading the code fwiw, so I imagine you might get some confusion here.

My point is just that these definitions seem easily confused and not very discoverable

My suggestion would be the following terminology convention (though I'm still not 100% sold on it):

- **Long name** Immutable unowned object
  - **For short** Permanent object 
- **Long name** Mutable unowned object
  - **For short** Shared object 

Again, not sold on these names, but I feel there should be no overlap on the words used on the long/short versions

Reply:
> I find all of this phrasing super confusing. It is really hard for me to follow if you say, We have "shared immutable" and "shared mutable", but we will just call them "immutable" and "shared" for short.
> 
> I found this doubly confusing while reading the code fwiw, so I imagine you might get some confusion here.
> 
> My point is just that these definitions seem easily confused and not very discoverable
> 
> My suggestion would be the following terminology convention (though I'm still not 100% sold on it):
> 
> * **Long name** Immutable unowned object
>   
>   * **For short** Permanent object
> * **Long name** Mutable unowned object
>   
>   * **For short** Shared object
> 
> Again, not sold on these names, but I feel there should be no overlap on the words used on the long/short versions

Agree here. I had already added this TODO to the same effect:
TODO: If immutable objects are considered shared by default, I'm not sure we can call shared mutable objects *shared objects* for short as we say above since both immutable and mutable can be shared.

Let's sort these out. I can then add to the glossary for quick reference. I'm good with Todd's suggestions for unowned objects.

Will owned objects always simply be that? Or are there variants there, as well, we should settle on the terminology for now?

Reply:
Good points. I agree this is confusing.
Regarding naming, let me come up with a few options and we can discuss in the channel